alf/trainers/policy_trainer.py

```diff
@@ -204,6 +204,13 @@ def progress():
 
     def train(self):
         """Perform training."""
+        self._checkpointer = Checkpointer(
+            ckpt_dir=os.path.join(self._train_dir, 'algorithm'),
+            algorithm=self._algorithm,
+            metrics=nn.ModuleList(self._algorithm.get_metrics()),
+            trainer_progress=self._trainer_progress)
+        if self._checkpointer.has_checkpoint():
+            self._algorithm.train_iter()
```

Comment:
Might be able to move Line 207-213 into the ```_restore_checkpoint()``` function, since they are all essentially part of this restoring process?
And it might be useful to add a note on the calling of ```self._algorithm.train_iter()``` for constructing any lazily created content related to checkpoint.

Reply:
good point